server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -315,23 +327,30 @@ private boolean showBlock(Block block) {
   }
 
   private boolean evaluateVisibility(Block block, PredicateDefinition predicate) {
-    JsonPathPredicateGenerator predicateGenerator =
-        new JsonPathPredicateGenerator(
-            this.programDefinition.streamQuestionDefinitions().collect(toImmutableList()),
-            block.getRepeatedEntity());
-    PredicateEvaluator predicateEvaluator =
-        new PredicateEvaluator(this.applicantData, predicateGenerator);
-
+    boolean evaluation = evaluatePredicate(block, predicate);
     switch (predicate.action()) {
       case HIDE_BLOCK:
-        return !predicateEvaluator.evaluate(predicate.rootNode());
+        return !evaluation;
       case SHOW_BLOCK:
-        return predicateEvaluator.evaluate(predicate.rootNode());
+        return evaluation;
       default:
         return true;
     }
   }
 
+  private boolean evaluateEligibility(Block block, PredicateDefinition predicate) {
+    return evaluatePredicate(block, predicate);
+  }
+
```

Comment:
suggestion (non-blocking): i could see leaving this for symmetry with `evaluateVisibility`, but since it's a private method I'm inclined to suggest removing it in favor of calling `evaluatePredicate` directly from `isBlockEligible`

Reply:
done